### PR TITLE
[ISSUE#5863]add nacos client batchRegitserInstance request

### PR DESCRIPTION
**Pull request by chenhao26-nineteen**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

for #5863 nacos client support batchRegister request

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by markguy** [comment]
received

File: client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/redo/RedoScheduledTask.java [hunk 1]
```diff
@@ -75,6 +76,12 @@ private void redoForInstance(InstanceRedoData redoData) throws NacosException {
                 if (isClientDisabled()) {
                     return;
                 }
+                if (redoData instanceof BatchInstanceRedoData) {
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/redo/RedoScheduledTask.java:79]: abstract line from 79 to 85 into other method.

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 2]
```diff
@@ -145,6 +147,26 @@ public static void checkInstanceIsLegal(Instance instance) throws NacosException
...
+    /**
+     * Batch verify the validity of instances.
+     * @param instances List of instances to be registered
+     * @throws NacosException Nacos
+     */
+    public static void batchCheckInstanceIsLegal(List<Instance> instances) throws NacosException {
+        CopyOnWriteArrayList<Instance> newInstanceList = new CopyOnWriteArrayList<>(instances);
+        for (int i = 0; i < newInstanceList.size(); i++) {
+            Instance instance = newInstanceList.get(i);
+            checkInstanceIsLegal(instance);
```
[the remark this answers is not in the document]
Comment (chenhao26-nineteen) [reply api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:172]: > should we check all instance is same ephemeral?

Yes, I adjusted it

File: client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/redo/RedoScheduledTask.java (hunk 1 above)
Comment (chenhao26-nineteen) [reply client/src/main/java/com/alibaba/nacos/client/naming/remote/gprc/redo/RedoScheduledTask.java]: > abstract line from 79 to 85 into other method.

i abstract it

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java [hunk 3]
```diff
@@ -145,6 +148,31 @@ public static void checkInstanceIsLegal(Instance instance) throws NacosException
...
     
+    /**
+     * check batch register is Ephemeral.
+     * @param instance instance
+     * @throws NacosException NacosException
+     */
+    public static void checkInstanceIsEphemeral(Instance instance) throws NacosException {
+        if (!instance.isEphemeral()) {
+            throw new NacosException(NacosException.INVALID_PARAM,
+                    String.format("Batch registration does not allow non-temporary instance registration , Instance：%s", instance));
```
Comment (KomachiSion) [inline api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:159]: `non-temporary` -> `non-ephemeral` or `persistent`

File: client/src/main/java/com/alibaba/nacos/client/naming/remote/http/NamingHttpClientProxy.java [hunk 4]
```diff
@@ -487,7 +492,7 @@ public String callServer(String api, Map<String, String> params, Map<String, Str
             HttpRestResult<String> restResult = nacosRestTemplate
                     .exchangeForm(url, header, Query.newInstance().initParams(params), body, method, String.class);
             end = System.currentTimeMillis();
-            
+
```
Comment (KomachiSion) [inline client/src/main/java/com/alibaba/nacos/client/naming/remote/http/NamingHttpClientProxy.java:495]: Why remove indent?

**Review by KomachiSion — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java (hunk 3 above)
Comment (chenhao26-nineteen) [reply api/src/main/java/com/alibaba/nacos/api/naming/utils/NamingUtils.java:159]: > persistent

rigorous

File: client/src/main/java/com/alibaba/nacos/client/naming/remote/http/NamingHttpClientProxy.java (hunk 4 above)
Comment (chenhao26-nineteen) [reply client/src/main/java/com/alibaba/nacos/client/naming/remote/http/NamingHttpClientProxy.java:494]: > Why remove indent?

I have revised it

**Review by KomachiSion — APPROVED** [review]

**Pull request merged**